### PR TITLE
docs: Add upgrade guide for Meilisearch plugin

### PR DESCRIPTION
**Pull request by olivermrbl**
Closes DOCS-668

### CODE REVIEW

**Comment by olivermrbl** [comment]
Exact version still to be decided. Please keep as draft until then :)

File: docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Meilisearch Search: v1.0.0
+
+Version 1.0.0 of our official Meilisearch plugin comes with breaking changes to the plugin options that are passed to the Meilisearch service through `medusa-config.js`.
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md:3]:
```suggestion
Version 1.0.0 of the official Meilisearch plugin comes with breaking changes to the plugin options that are passed to the Meilisearch service through `medusa-config.js`.
```

File: docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      indexSettings: {
+        searchableAttributes: string[]
+        displayedAttributes: string[]
+				...
+      }
+      transformer?: <T>(medusaEntity: T) => unknown
+    }
+  }
+}
+```
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md:28]: As we have ESLint configured on languages like `js`, `jsx`, `ts`, and `tsx`, this block produces errors related to JavaScript. I suggest that we change the block to the following:

~~~suggestion
```js title=medusa-config.js
const plugins = [
  // ...
  {
    config: {
      host,
      apiKey
    }
    settings: {
      indexName: {
        indexSettings: {
          searchableAttributes,
          displayedAttributes
          // ...
        }
        transformer?: (product) => ({
          id: product.id,
          // ...
        })
      }
    }
  }
]
```
~~~

And then we can link to the MeiliSearch plugin for more details on what each of these mean, although I don't think that's necessary considering you already cover it later in the upgrade guide.

File: docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+## Actions required
+
+As you can see from the new object shape, the property `indexSettings` has been introduced to hold the settings specific to Meilisearch’s index options. This has been done to make space for the `transformer` and `primaryKey`.
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md:32]:
```suggestion
As you can see from the new object shape, the property `indexSettings` has been introduced to hold the settings specific to Meilisearch’s index options. This has been done to make space for new settings like `transformer` and `primaryKey`. 
```

File: docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  }
+}
+```
+
+## Actions required
+
+As you can see from the new object shape, the property `indexSettings` has been introduced to hold the settings specific to Meilisearch’s index options. This has been done to make space for the `transformer` and `primaryKey`. 
+
+Previously, you might have configured the Meilisearch plugin as seen below:
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md:34]:
```suggestion
Previously, your Meilisearch plugin configurations were something like this:
```

File: docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    apiKey: "some_key",
+	}
+  settings: {
+    products: {
+      searchableAttributes: ["title", "description"],
+      attributesToRetrieve: ["title", "description"],
+		}
+  }
+}
+```
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md:49]:
~~~suggestion
```js title=medusa-config.js
const plugins = [
  // ...
  {
    config: {
      host: "https://search-api-example.com",
      apiKey: "some_key",
    },
    settings: {
      products: {
        searchableAttributes: ["title", "description"],
        attributesToRetrieve: ["title", "description"],
      }
    }
  }
]
```
~~~

File: docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+In the above example, an index `products` has been configured with two options. Updating to 1.0.0 requires you to nest these within the `indexSettings`.
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md:51]:
```suggestion
In the above example, an index `products` has been configured with the two options `searchableAttributes` and `attributesToRetrieve`. Updating to 1.0.0 requires you to nest these options within the `indexSettings`. 
```

File: docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  settings: {
+    products: {
+			indexSettings: {
+	      searchableAttributes: ["title", "description"],
+	      attributesToRetrieve: ["title", "description"],	
+			}
+		}
+  }
+}
+```
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md:70]:
~~~suggestion
```js
const plugins = [
  // ...
  {
    config: {
      host: "https://search-api-example.com",
      apiKey: "some_key",
    },
    settings: {
      products: {
        indexSettings: {
          searchableAttributes: ["title", "description"],
          attributesToRetrieve: ["title", "description"],	
        }
      }
    }
  }
]
```
~~~

File: docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+You can read more about the new plugin options properties in the package README.
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/meilisearch/1-0-0.md:72]:
```suggestion
You can learn more about the new settings in the [MeiliSearch plugin documentation](../../../plugins/search/meilisearch.md)
```

**Review by shahednasser — CHANGES_REQUESTED** [review]
Left some suggestions for fixes. Can you also run `yarn lint:docs --fix` in the root directory of the monorepo to fix eslint errors?

**Review by shahednasser — APPROVED** [review]
LGTM! Thanks a lot for your work @olivermrbl 🚀

**Pull request merged**